### DISCUSSION TITLE
Weird livewire related error during CI deployment via Bitbucket Pipelines

### DISCUSSION THREAD

**Discussion by Mushoz** [livewire/livewire#7720] [1. Help] [answered]
I am using a bitbucket pipeline to deploy a Filament project to our webserver. The final step in this pipeline is using Atlassian's ssh-run pipe (https://bitbucket.org/atlassian/ssh-run/src/master/) to run a bash script that will deploy the installation, that was uploaded in a previous step through scp as a zip-file.

Now the output of this ssh-run is the following:

```
artisan down

PHP Fatal error:  Constant expression contains invalid operations in /projects/mentrix_9aff/public_html/vendor/livewire/livewire/src/helpers.php on line 120

artisan config

PHP Fatal error:  Constant expression contains invalid operations in /projects/mentrix_9aff/public_html/vendor/livewire/livewire/src/helpers.php on line 120

artisan event

PHP Fatal error:  Constant expression contains invalid operations in /projects/mentrix_9aff/public_html/vendor/livewire/livewire/src/helpers.php on line 120

artisan route

PHP Fatal error:  Constant expression contains invalid operations in /projects/mentrix_9aff/public_html/vendor/livewire/livewire/src/helpers.php on line 120

artisan view

PHP Fatal error:  Constant expression contains invalid operations in /projects/mentrix_9aff/public_html/vendor/livewire/livewire/src/helpers.php on line 120

artisan migrate

PHP Fatal error:  Constant expression contains invalid operations in /projects/mentrix_9aff/public_html/vendor/livewire/livewire/src/helpers.php on line 120

artisan up

PHP Fatal error:  Constant expression contains invalid operations in /projects/mentrix_9aff/public_html/vendor/livewire/livewire/src/helpers.php on line 120
```

This output is generated by the following commands:

```
echo "artisan down"
php artisan down

...

echo "artisan config"
php artisan config:cache
echo "artisan event"
php artisan event:cache
echo "artisan route"
php artisan route:cache
echo "artisan view"
php artisan view:cache
echo "artisan migrate"
php artisan migrate --force
echo "artisan up"
php artisan up
```

In other words, each time php artisan gets called, this error is thrown. The pipeline itself succeeds, and the program is successfully deployed. But these last few updates have had no migrations included, so the fact the application works is not indicative that the artisan commands themselves were actually successful.

Strangely, if I ssh into the server manually from the terminal, and run the EXACT same bash script in my own terminal, the artisan commands all complete successfully, and no such error is thrown.

While it does work properly, I am not too happy to ignore these errors without knowing why there are there, and whether they are harmless or not. So I was hoping anyone had some pointers on what could be happening. Thanks!

**Comment by Mushoz** [comment]
Quick update: I have added a `sleep 60` after the artisan down command, and I can confirm that my application is NOT down while it's sleeping. So the artisan commands themselves really are not doing anything at all at the moment.

Running the same bash script from a SSH terminal DOES show the application as down while it's sleeping. So the artisan commands themselves really are failing in our CI pipeline.

**Comment by Mushoz** [comment, marked as answer]
Found the solution myself fortunately! Apparently, the third party hosting provider has an alias as such:

`alias php=php8.2'

This alias is active when I SSH into the server, which is why the bash deployment script that utilizes the php command works fine, as it's a php 8.2 project.

However, Atlassian's ssh-run docker image doesn't log in via SSH, but runs the bash script directly in a one-liner via `ssh user@server 'bash s' < bashscript.sh`

This does NOT use the server's alias, but the php command is still valid, since it's the binary for php 8.0. But obviously that makes the artisan commands fail, as it's now using the incorrect php version.
